src/helpers/helpers.collection.js

```diff
@@ -35,7 +35,10 @@ export function _lookup(table, value, cmp) {
  */
 export const _lookupByKey = (table, key, value, last) =>
   _lookup(table, value, last
-    ? index => table[index][key] <= value
+    ? index => {
+      const ti = table[index][key];
+      return ti < value || ti === value && table[index + 1][key] === value;
```

Comment:
Can this crash if `index+1` goes out of bounds?

Reply:
https://github.com/chartjs/Chart.js/blob/master/src/helpers/helpers.collection.js#L16-L18

@etimberg as I see `index` is never last index